### PR TITLE
feature: d2p support take over docker's running containers

### PR DESCRIPTION
**Pull request by HusterWan**
Signed-off-by: Michael Wan <zirenwan@gmail.com>

**Note:**   we must use alibaba containerd to support taking over docker's running containers when upgrade docker to PouchContainer.

### CODE REVIEW

File: ctrd/containerd.go [hunk 1]
```diff
@@ -246,3 +246,27 @@ func (ctrd *Ctrd) RemoveSnapshot(ctx context.Context, id string) error {
 	return service.Remove(ctx, id)
 
 }
+
+// NewContainer just create a very simple container for migration
+// just load container id to boltdb
+func (ctrd *Ctrd) NewContainer(ctx context.Context, id string) error {
+	ctx = namespaces.WithNamespace(ctx, namespaces.Default)
+
+	spec, err := oci.GenerateSpec(ctx, nil, &containers.Container{ID: id})
```
Comment (Letty5411) [inline ctrd/containerd.go]: Do we need to remove the containerID info if any step failed?

File: main.go [hunk 2]
```diff
@@ -72,6 +74,8 @@ func setupFlags(cmd *cobra.Command) {
 	flagSet.StringVar(&cfg.PouchPkgPath, "pouch-pkg-path", "pouch", "Specify pouch package file path")
 	flagSet.BoolVar(&cfg.MigrateAll, "migrate-all", false, "If true, do all migration things, otherwise, just prepare data for migration")
 	flagSet.BoolVarP(&cfg.Debug, "debug", "D", false, "DEBUG mode log level")
+	flagSet.BoolVar(&cfg.DryRun, "dry-run", false, "we will not remove docker package, if dry-run flag set")
```
Comment (Letty5411) [inline main.go]: Is this flag conflict with any other flag?

File: ctrd/containerd.go (hunk 1 above)
Comment (HusterWan) [reply ctrd/containerd.go]: make sense, i will fix

File: main.go (hunk 2 above)
Comment (HusterWan) [reply main.go]: no! `dry-run` flag just use for test

**Comment by HusterWan** [comment]
@Letty5411 updated ! PTAL thanks

File: main.go [hunk 3]
```diff
@@ -72,6 +74,8 @@ func setupFlags(cmd *cobra.Command) {
...
+	flagSet.BoolVar(&cfg.AutoTakeoverDockerContainer, "auto-takeover-docker-container", false, "auto takeover docker container when migrate Docker to PouchContainer")
```
Comment (allencloud) [inline main.go]: `auto takeover docker running containers which are under alibaba/containerd 0.2.4 when migrating Docker to PouchContainer`

**Comment by allencloud** [comment]
LGTM

**Pull request merged**